napari/components/add_layers_mixin.py

```diff
@@ -45,6 +50,57 @@ def add_layer(self, layer):
         if len(self.layers) == 1:
             self.reset_view()
 
+    def add_path(
+        self,
+        path: Union[str, Sequence[str]],
+        stack: bool = False,
+        use_dask: Optional[bool] = None,
+    ) -> List[layers.Layer]:
+        """Add a path to the viewer
+
+        Paths will be handed one-by-one to the napari_get_reader hook.
+
+        Parameters
+        ----------
+        path : str or list of str
+            path(s) to view.
+        stack : bool, optional
+            Concatenate multiple input files into a single stack,
+            by default False
+        use_dask : bool, optional
+            Whether to use dask to create a lazy array, rather than NumPy.
+            If ``None``, will resolve to True if filenames contains more than
+            one image, False otherwise.  by default None.
+        """
+        paths = [path] if isinstance(path, str) else path
```

Comment:
someone has before asked for pathobjects to be used in #708 so we added https://github.com/napari/napari/blob/master/napari/utils/io.py#L38-L40 to magic_imread, should we have those lines here too / update docs accordingly

Reply:
Yes this is a good idea. =)

Reply:
yes I agree... will do.  curious though, why did you use `path.as_posix()` instead of `str(path)`?  is that going to be as robust on windows?

Reply:
iirc we checked that `as_posix` works on Windows. 🤷‍♂ I'm old fashioned and don't use Pathlib, so I can't say too much about it... =)

Reply:
I can't say for sure - if I wrote that line then I must have stackover flow that made the suggestion, but i'm really willing to defer to others who know more than i do here